sdks/python/apache_beam/runners/worker/statesampler_test.py

```diff
@@ -24,40 +24,51 @@
 import unittest
 from builtins import range
 
+from tenacity import retry
+from tenacity import stop_after_attempt
+
 from apache_beam.runners.worker import statesampler
 from apache_beam.utils.counters import CounterFactory
 from apache_beam.utils.counters import CounterName
 
 
 class StateSamplerTest(unittest.TestCase):
 
+  # Due to somewhat non-deterministic nature of state sampling and sleep,
+  # this test is flaky when state duraiton is low.
+  # Since increasing state duration significantly would als slow down
+  # the test suite, we are retrying once on failure as a mitigation.
+  @retry(reraise=True, stop=stop_after_attempt(2))
```

Comment:
Does this try a total of 2 or 3 times? I recommend 3 if that is not the case already.

Reply:
This gives the test two attempts total to pass. Originally, I intentionally set a low amount of retries, in case there is some issue we may want to catch, related to original purpose of this test. Increasing to 3.